oracle/src/events/processAMBCollectedSignaturesMEV/index.js

```diff
@@ -0,0 +1,174 @@
+const { signatureToVRS, packSignatures } = require('../../utils/message')
+const { readAccessListFile, isRevertError } = require('../../utils/utils')
+const { parseAMBMessage } = require('../../../../commons')
+const estimateGas = require('../processAMBCollectedSignatures/estimateGas')
+const { AlreadyProcessedError, IncompatibleContractError, InvalidValidatorError } = require('../../utils/errors')
+const { MAX_CONCURRENT_EVENTS, EXTRA_GAS_ABSOLUTE } = require('../../utils/constants')
+
+const limit = promiseLimit(MAX_CONCURRENT_EVENTS)
+
+const { ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST, ORACLE_HOME_TO_FOREIGN_BLOCK_LIST } = process.env
+
+function processCollectedSignaturesBuilder(config) {
+  const { home, foreign, mevForeign } = config
+
+  let validatorContract = null
+
+  return async function processCollectedSignatures(signatures) {
+    const txToSend = []
+
+    if (validatorContract === null) {
+      validatorContract = await getValidatorContract(foreign.bridgeContract, foreign.web3)
+    }
+
+    rootLogger.debug(`Processing ${signatures.length} CollectedSignatures events`)
+    const callbacks = signatures
+      .map(colSignature => async () => {
+        const { messageHash, NumberOfCollectedSignatures } = colSignature.returnValues
+
+        const logger = rootLogger.child({
+          eventTransactionHash: colSignature.transactionHash
+        })
+
+        logger.info(`Processing CollectedSignatures ${colSignature.transactionHash}`)
+        const message = await home.bridgeContract.methods.message(messageHash).call()
+        const parsedMessage = parseAMBMessage(message)
+
+        if (ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST || ORACLE_HOME_TO_FOREIGN_BLOCK_LIST) {
+          const sender = parsedMessage.sender.toLowerCase()
+          const executor = parsedMessage.executor.toLowerCase()
+
+          if (ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST) {
+            const allowanceList = await readAccessListFile(ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST, logger)
+            if (!allowanceList.includes(executor) && !allowanceList.includes(sender)) {
+              logger.info(
+                { sender, executor },
+                'Validator skips a message. Neither sender nor executor addresses are in the allowance list.'
+              )
+              return
+            }
+          } else if (ORACLE_HOME_TO_FOREIGN_BLOCK_LIST) {
+            const blockList = await readAccessListFile(ORACLE_HOME_TO_FOREIGN_BLOCK_LIST, logger)
+            if (blockList.includes(executor)) {
+              logger.info({ executor }, 'Validator skips a message. Executor address is in the block list.')
+              return
+            }
+            if (blockList.includes(sender)) {
+              logger.info({ sender }, 'Validator skips a message. Sender address is in the block list.')
+              return
+            }
+          }
+        }
+
+        logger.debug({ NumberOfCollectedSignatures }, 'Number of signatures to get')
```

Comment:
It make sense to check if the message was marked to go through the manual lane like it was done here: https://github.com/poanetwork/tokenbridge/blob/b17fff2b5678adf0272329870d7ade8927f16696/oracle/src/events/processAMBCollectedSignatures/index.js#L79-L85

Reply:
I think that it is useful to consider manual messages, since it makes sense to the oracle to execute them as well, if they are profitable for it.

Reply:
I think we need to be consistent here:
- If the allowance or block lists are configured we skips the message even if it could be profitable.
- But if the message is sent to the manual line by the contract we try to estimate if it is profitable or not. 

What if we check whether the sender/receiver of the message is within the allowance list and only if it is true, check the lane the message was sent through? Or at least to make the second check configurable. I just would like to have an ability to not estimate profit for messages I do not intend to execute.